File: Classes/PSBookmark.m
//
//  PSBookmark.m
//  PocketSword
//
//  Created by Nic Carter on 5/01/11.
//  Copyright 2011 CrossWire Bible Society. All rights reserved.
//

#import "PSBookmark.h"


@implementation PSBookmark

@synthesize ref;

- (id)initWithName:(NSString *)n dateAdded:(NSDate *)da dateLastAccessed:(NSDate *)dla bibleReference:(NSString*)r {
	self = [super initWithName:n dateAdded:da dateLastAccessed:dla];
	if(self) {
		self.ref = r;
	}
	return self;
}

- (void)dealloc {
	self.ref = nil;
	[super dealloc];
}



@end

File: Classes/PSBookmark.swift
//
//  PSBookmark.swift
//  PocketSword
//
//  Swift migration step 1.1 — bookmark inheritance chain (ATOMIC, §2A Rule 3).
//  Port of PSBookmark.{h,m}.
//

import Foundation

@objc(PSBookmark)
class PSBookmark: PSBookmarkObject {

    @objc var ref: String?

    @objc init(name n: String?, dateAdded da: Date?, dateLastAccessed dla: Date?, bibleReference r: String?) {
        super.init(name: n, dateAdded: da, dateLastAccessed: dla)
        self.ref = r
    }
}
